chain/client/src/client.rs

```diff
@@ -1155,6 +1148,10 @@ impl Client {
             &mut block_processing_artifacts,
             apply_chunks_done_callback,
         );
+        if accepted_blocks.iter().any(|accepted_block| accepted_block.status.is_new_head()) {
```

Comment:
I've forgotten to add code to update header_head when it is updated due to header sync. Reminder to self.